src/sesclient.native/sesclient.native.cpp

```diff
@@ -3,89 +3,167 @@
-    std::cout << "Contacts the specified Azure Batch software entitlement server to verify the provided token." << std::endl;
-    std::cout << "Parameters:" << std::endl;
-    std::cout << "    --url <software entitlement server URL>" << std::endl;
-    std::cout << "    --thumbprint <thumbprint of a certificate expected in the server's SSL certificate chain>" << std::endl;
-    std::cout << "    --common-name <common name of the certificate with the specified thumbprint>" << std::endl;
-    std::cout << "    --token <software entitlement token to pass to the server>" << std::endl;
-    std::cout << "    --application <name of the license ID being requested>" << std::endl;
+	std::cout << exeName << ":" << std::endl;
+	std::cout << "Contacts the specified Azure Batch software entitlement server to verify the provided token." << std::endl;
+	std::cout << "Parameters:" << std::endl;
+	std::cout << "    --url <software entitlement server URL>" << std::endl;
+	std::cout << "    --thumbprint <thumbprint of a certificate expected in the server's SSL certificate chain (optional)>" << std::endl;
+	std::cout << "    --common-name <common name of the certificate with the specified thumbprint (optional)>" << std::endl;
+	std::cout << "    --token <software entitlement token to pass to the server>" << std::endl;
+	std::cout << "    --application <name of the license ID being requested>" << std::endl;
 }
 
 
-static const std::array<std::string, 5> parameterNames = {
-    "--url",
-    "--thumbprint",
-    "--common-name",
-    "--token",
-    "--application"
+static const std::vector<std::string> mandatoryParameterNames = {
+	"--url",
+	"--token",
+	"--application"
 };
 
+static const std::vector<std::string> optionalParameterNames = {
+	"--thumbprint",
+	"--common-name"
+};
 
 struct Initializer
 {
-    Initializer()
-    {
-        int err = Microsoft::Azure::Batch::SoftwareEntitlement::Init();
-        if (err != 0)
-        {
-            throw std::runtime_error(
-                "Microsoft::Azure::Batch::SoftwareEntitlement::Init failed with error " +
-                std::to_string(err)
-            );
-        }
-    }
-
-    ~Initializer()
-    {
-        Microsoft::Azure::Batch::SoftwareEntitlement::Cleanup();
-    }
+	Initializer()
+	{
+		int err = Microsoft::Azure::Batch::SoftwareEntitlement::Init();
+		if (err != 0)
+		{
+			throw std::runtime_error(
+				"Microsoft::Azure::Batch::SoftwareEntitlement::Init failed with error " +
+				std::to_string(err)
+			);
+		}
+	}
+
+	~Initializer()
+	{
+		Microsoft::Azure::Batch::SoftwareEntitlement::Cleanup();
+	}
 };
 
+auto shouldShowUsage = false;
+auto configurationError = false;
+
+void CheckForMandatoryParameters(const std::unordered_map<std::string, std::string>& parameters)
+{
+	for (const auto& param : mandatoryParameterNames)
+	{
+		if (parameters.find(param) == parameters.end())
+		{
+			std::cout << "Missing mandatory parameter " << param << std::endl;
```

Comment:
You could either throw here, or build a string and if non-empty at the end of the for loop, throw that.